tutorials/grid_coordinates.py

```diff
@@ -0,0 +1,289 @@
+"""
+.. _grid_coordinates:
+
+Grid coordinates
+====================
+
+Grid coordinates in Verde are used to create points on a regularly spaced grid
+that are then used in the spline method to interpolate between sample data
+points. The grid can be specified either by the number of points in each
+dimension (the *shape*) or by the grid node spacing.
+
+Creating regular grids in Verde is done using the :func:`verde.grid_coordinates`
+function. It creates a set of regularly spaced points in both the west-east and
+south-north directions, i.e. a two-dimensional spatial grid.
+
+"""
+
+import numpy as np
+import matplotlib.pyplot as plt
+import verde as vd
+from matplotlib.patches import Rectangle
+
+########################################################################################
+# First let's create a region that is 1000 units west-east and 1000 units south-north,
+# and we will set an initial spacing to 100 units.
+
+spacing = 100
+west, east, south, north = 0, 1000, 0, 1000
+region = (west, east, south, north)
+
+# create the grid coordinates
+easting, northing = vd.grid_coordinates(region=region, spacing=spacing)
+
+########################################################################################
+# We can check the dimensions of the grid coordinates to confirm that it is 11x11
+
+print(easting.shape, northing.shape)
+
+########################################################################################
+# Let's define two functions to visualize the region bounds and grid points
+
+
+def plot_region(ax, region=None):
+    "Plot the region as a solid line."
+    if region is not None:
+        west, east, south, north = region
+        ax.add_patch(
+            plt.Rectangle((west, south), east, north, fill=None, label="Region Bounds")
+        )
+
+
+def plot_grid(ax, coordinates, linestyles="dotted", region=None, pad=50, **kwargs):
+    "Plot the grid coordinates as dots and lines."
+    data_region = vd.get_region(coordinates)
+    ax.vlines(
+        coordinates[0][0],
+        ymin=data_region[2],
+        ymax=data_region[3],
+        linestyles=linestyles,
+        zorder=0,
+    )
+    ax.hlines(
+        coordinates[1][:, 1],
+        xmin=data_region[0],
+        xmax=data_region[1],
+        linestyles=linestyles,
+        zorder=0,
+    )
+    ax.scatter(*coordinates, **kwargs)
+    if pad:
+        padded = vd.pad_region(region, pad=pad)
+        plt.xlim(padded[:2])
+        plt.ylim(padded[2:])
+
+
+# Visualize our region and grid coordinates using our functions
```

Comment:
Maybe add a `############...` to make this render as rst instead of a comment?

Reply:
I like it as rst better than an inline comment. Looks much better!